packages/shared/lib/sdk/sync.ts

```diff
@@ -757,3 +759,59 @@ const persistApi = axios.create({
         return true;
     }
 });
+
+const TELEMETRY_ALLOWED_METHODS: (keyof NangoSync)[] = [
+    'batchDelete',
+    'batchSave',
+    'batchSend',
+
+    'getConnection',
+    'getEnvironmentVariables',
+    'getMetadata',
+
+    'proxy',
+
+    'log'
+];
+
+/* eslint-disable no-inner-declarations */
+/**
+ * This function will enable tracing on the SDK
+ * It has been split from the actual code to avoid making the code too dirty and to easily enable/disable tracing if there is an issue with it
+ */
+export function instrumentSDK(rawNango: NangoAction | NangoSync, tracer: Tracer) {
+    return new Proxy(rawNango, {
+        get<T extends typeof rawNango, K extends keyof typeof rawNango>(target: T, propKey: K) {
+            // Method name is not matching the allowList we don't do anything else
+            if (!TELEMETRY_ALLOWED_METHODS.includes(propKey)) {
+                return target[propKey];
+            }
+
+            if (propKey === 'proxy') {
+                // In case of Proxy we want to log what website we are reaching
+                function proxified(this: T, ...args: Parameters<NangoAction['proxy']>) {
+                    const scope = tracer.scope().active();
+                    scope?.addTags({
+                        proxy: { method: args[0].method, endpoint: args[0].endpoint }
+                    });
+                    return target[propKey as 'proxy'].apply(this, args);
+                }
+                return tracer.wrap(`${SpanTypes.RUNNER_SDK}.${propKey}`, proxified);
+            } else if ((propKey as keyof NangoSync) === 'batchSave') {
+                // In case of BatchSave the number of records are important
+                function proxified(this: NangoSync, ...args: Parameters<NangoSync['batchSave']>) {
```

Comment:
we have this info in persistAPI though. do we need to add it here as well?

Reply:
it's debatable, I added more than just the proxy because there are two different duration. 
The one observed from the Persist API and the one observed from the client. Persist can be fast but if the client has some network issue they won't have the timing. 
wdyt?

Reply:
ah yes, I just to get a sense of why the response could be slow. If they send 10K records in a batchSave for example that could be helpful